### ISSUE TITLE
About dataset formats and DCT codes.

### ISSUE THREAD

**Issue by cyber0777** [alexhe101/FAME-Net#7]
Thanks to the author for the excellent work. I have three queries that I would like to get your response on.
1. Referring to data/dataset.py in the repo, the inputs ms/pan are the whole images and not the crop images? Could you give an example of a dataset shape?
2. Can you provide an example of a DCT-generated mask? Referring to Figure 1 in the paper, is the mask obtained by processing HRMS?
3. Finally, referring to the Class Data in data/dataset.py, the bms_image (mask) is obtained by upsampling the lms_image, which seems independent of the mask generated in advance.

These are all my questions and I would be grateful if the author could reply! Thanks.

**Comment by alexhe101** [comment]
Hi,

1. The images in our dataset are cropped already, so there’s no additional cropping in the code. The shape of the MS images is (128,128,4), PAN is (128,128), and LRMS is (32,32,4).

2. The masks are generated either from LRMS or PAN, both are valid. The process is as follows for reference:
- Convert the LRMS image to grayscale, then apply DCT to transform it into the frequency domain.
- Apply the mask in the frequency domain, and then transform it back to the spatial domain.
- Afterward, binarize the image by setting very low pixel values to zero, which may leave some noise.
- Apply median filtering to remove this noise, resulting in the mask shown in Figure 1 of the paper.

3. The bms_image refers to the bicubic upsampled image, not the mask.

**Comment by cyber0777** [comment]
Thank you for your reply.

1. In line 138 in sovler/moesolver of repo, I noticed that the supervision information of the mask, mask_gt, corresponds to bms_image in data/dataset.py. This is the reason for the third question.
2. Additionally, I would like to know, if the size of LRMS is (32, 32, 4) and PAN is (128, 128), will the generated mask be (128, 128, 2)?
3. What is the directory for the generated mask files? 

I look forward to your reply again!

**Comment by alexhe101** [comment]
1. Regarding the code at [line 141](https://github.com/alexhe101/FAME-Net/blob/main/data/dataset.py), here’s the corrected explanation:
```python
if self.mask_image_filenames:
    hf = mask_image  ## I use bms_image here, but it should be mask_image
    lf = 1 - hf
    bms_image = torch.cat([hf, lf], dim=0)
return ms_image, lms_image, pan_image, bms_image, file
```
2. No, the generated mask will have only 1 channel.
3. The dataset directory can be organized as follows:
- MS: dataset/wv2/train128/ms/1.tif, 2.tif, ... n.tif
- PAN: dataset/wv2/train128/pan/1.tif, 2.tif, ... n.tif
- Mask: dataset/wv2/train128/mask/1.png, 2.png, ... n.png

**Comment by cyber0777** [comment]
Thank you very much for your explanation！

**Comment by violet2422** [comment]
> Hi,
> 
> 1. The images in our dataset are cropped already, so there’s no additional cropping in the code. The shape of the MS images is (128,128,4), PAN is (128,128), and LRMS is (32,32,4).
> 2. The masks are generated either from LRMS or PAN, both are valid. The process is as follows for reference:
> 
> * Convert the LRMS image to grayscale, then apply DCT to transform it into the frequency domain.
> * Apply the mask in the frequency domain, and then transform it back to the spatial domain.
> * Afterward, binarize the image by setting very low pixel values to zero, which may leave some noise.
> * Apply median filtering to remove this noise, resulting in the mask shown in Figure 1 of the paper.
> 
> 3. The bms_image refers to the bicubic upsampled image, not the mask.

如果不方便提供代码，有没有相关论文代码作为参考呢？

**Comment by violet2422** [comment]
> Thank you very much for your explanation！

您好！请问您复现了mask生成的代码了吗？如果准确复现，可以提供给我，感激不尽！！！
